### DISCUSSION TITLE
Update typ Header Parameter to support RFC 9068's validating access token recommendations

### DISCUSSION THREAD

**Discussion by JonSzeto821** [keycloak/keycloak#19419] [Ideas]
Hello,

My team is currently working towards standardizing JWT tokens and the information encapsulated within them for RHSSO/Keycloak. We came across RFC 9068 and the proposed approach for validating JWT access tokens ([RFC 9068 section 4](https://www.rfc-editor.org/rfc/rfc9068.html#name-validating-jwt-access-token)). Keycloak's access token validation meets five of the six criteria suggested in RFC 9068. The validation that is currently missing is:

> _The resource server MUST verify that the "typ" header value is "at+jwt" or "application/at+jwt" and reject tokens carrying any other value._

To get Keycloak more aligned with the standards proposed by IETF, we would like to recommend the following change:

1. Change the `typ` Header Parameter value to `at+jwt` for access tokens
	- The current value for `typ` Header Parameter on access tokens is `JWT`
	- The value, `application/at+jwt`, is not being considered due to this RFC 7515: https://www.rfc-editor.org/rfc/rfc7515#section-4.1.9

Looking through past discussions, there appears to be support for being aligned with RFC 9068 here: https://github.com/keycloak/keycloak/discussions/8646

cc: @jsorah

**Comment by laurids** [comment]
Hi @JonSzeto821 
From version 26.2, it will be possible to set the access token header type to "at+jwt" via the client configuration. See https://github.com/keycloak/keycloak/pull/37573
